plugins/inputs/powerdns/powerdns.go

```diff
@@ -49,59 +49,105 @@ func (p *Powerdns) Gather(acc telegraf.Accumulator) error {
 }
 
 func (p *Powerdns) gatherServer(address string, acc telegraf.Accumulator) error {
-	conn, err := net.DialTimeout("unix", address, defaultTimeout)
+	stats_cmd := "show * \n"
+	// fetch general stats
+	stats, err := p.executeCommand(stats_cmd, address)
 	if err != nil {
 		return err
 	}
+	// Process general stats data
+	fields := parseStatsResponse(stats)
+	// Add server socket as a tag
+	tags := map[string]string{"server": address}
+	acc.AddFields("powerdns", fields, tags)
 
-	defer conn.Close()
+	// fetch respsizes stats for calculating histogram metrics
+	respsizes_cmd := "respsizes \n"
+	respsizes, err := p.executeCommand(respsizes_cmd, address)
+	if err != nil {
+		return err
+	}
+	// Process latency stats data
+	latencyBucket := parseRespsizesResponse(respsizes)
+	for latency, count := range latencyBucket {
+		tags["latency"] = strconv.Itoa(int(latency))
```

Comment:
Would it be possible to use the `le` tag like we used in the histogram aggregator and was inspired by [prometheus](https://prometheus.io/docs/concepts/metric_types/#histogram)?  This would help make our histograms more uniform.

Always use strconv.ParseInt to avoid issues on 32-bit operating systems, and make sure to not to convert to int for the same reason.